app/Services/UserService.php

```diff
@@ -64,6 +65,11 @@ public function loginUser($data, $authType)
             return response()->json(['error' => 'Email not verified. Please check your inbox.'], 403);
         }
 
+        if ($user->is_2fa_enabled) {
```

Comment:
make this code into a seperate function for example change the verifyOTP service into is2faEnabled where as parameter you send the user and the otp than inside the function you check if the user has the 2fa enabled if it has than you check if you have passed an otp code if you have passed an otp code you verify if the otp coce is correct if not than you call the function generateOTP